src/VisualStudio/LiveShare/Impl/Client/LocalForwarders/VBRemoteCompilationFactoryService.cs

```diff
@@ -11,9 +13,11 @@ namespace Microsoft.VisualStudio.LanguageServices.LiveShare.Client.LocalForwarde
     [ExportLanguageServiceFactory(typeof(ICompilationFactoryService), StringConstants.VBLspLanguageName), Shared]
     internal class VBRemoteCompilationFactoryService : ILanguageServiceFactory
     {
-        public ILanguageService CreateLanguageService(HostLanguageServices languageServices)
+        public ILanguageService? CreateLanguageService(HostLanguageServices languageServices)
         {
-            return languageServices.GetOriginalLanguageService<ICompilationFactoryService>();
+            // Don't allow the remote workspace to create VB compilations; since we don't have references, any attempt to use semantics in the workspace
+            // on the client side is incorrect.
+            return null;
```

Comment:
instead of having it return null, could we have it return a NoOp compilation factory?

Reply:
What would the no-op one do?

Reply:
REturn empty, or null compilations.  i think i'd like to isolate the nullness to that specific API, versus be at thsi higher abstract-service level.

Reply:
So empty would be bad since that'd violate the "document syntax tree is in compilation" being discussed in the other bug. Returning null here would mean you get null compilations., and Project.SupportsCompilation would say false.

Reply:
>  Returning null here would mean you get null compilations., and Project.SupportsCompilation would say false.

Isn't that true?  Since we don't have a CompilationFactory... we can't have compilations.  so i would expect these project did not support compilations.

Isi that not the case here?  If so... that also seems very weird.

Reply:
n/m  i see what you're saying.  i don't have a solution i love more than this.

Reply:
Well, we're not sure we actually want to do this. (This is a draft.) This is actually what Live Share *was* doing it turns out, and @dibarbet reminded us we added a compilation because otherwise other code was crashing. So this might get killed. We're not sure yet.